modules/environment/default.nix

```diff
@@ -193,6 +198,21 @@ in
       ${cfg.extraInit}
     '';
 
+    system.build.setEnvironmentFish = pkgs.writeText "set-environment-fish" ''
+      # Prevent this file from being sourced by child shells.
+      set -g -x __NIX_DARWIN_SET_ENVIRONMENT_DONE 1
+
+      set -g -x PATH ${config.environment.systemPath}
+      ${concatStringsSep "\n" exportVariablesFish}
+
+      # Extra initialisation
+      set fish_function_path ${pkgs.fish-foreign-env}/share/fish-foreign-env/functions $__fish_datadir/functions
+      fenv "source /etc/fish/foreign-env/extraInit" > /dev/null
```

Comment:
There's a couple of ways we could avoid fenv entirely: duplicate the `extraInit` in fish or preprocess during build and output the result here. I experimented with that [here](https://github.com/kevingriffin/plugin-foreign-env/commit/9e92894586ef3e49d03fbcedfcae311e0e75fada), but I realized that making fenv lazy presents an interesting challenge with interpolation: does it get deferred? If we chose that route, it'd be possible to modify fenv to support this use case.